ui/server.go

```diff
@@ -20,45 +20,98 @@ package ui
 import (
 	"context"
 	"fmt"
+	`net`
 	"net/http"
+	`net/http/httputil`
 	"strconv"
+	`strings`
 	"time"
 
 	log "github.com/cihub/seelog"
 	"github.com/gin-contrib/cors"
 	"github.com/gin-gonic/gin"
 	godvpnweb "github.com/mysteriumnetwork/go-dvpn-web"
-	"github.com/mysteriumnetwork/node/ui/discovery"
 	"github.com/pkg/errors"
+
+	"github.com/mysteriumnetwork/node/ui/discovery"
 )
 
 const logPrefix = "[dvpn-web-server] "
+const tequilapiUrlPrefix = "/tequilapi"
+const tequilapiHost = "127.0.0.1"
 
 // Server represents our web UI server
 type Server struct {
 	srv       *http.Server
 	discovery discovery.LANDiscovery
 }
 
+// ReverseTequilapiProxy proxies UIServer requests to the TequilAPI server
+func ReverseTequilapiProxy(tequilapiPort int) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// skip non Tequilapi routes
+		if !strings.Contains(c.Request.URL.Path, tequilapiUrlPrefix) {
+			return
+		}
+
+		proxy := &httputil.ReverseProxy{
+			Director: func(req *http.Request) {
+				req.URL.Scheme = "http"
+				req.URL.Host = tequilapiHost + ":" + strconv.Itoa(tequilapiPort)
+				req.URL.Path = strings.Replace(req.URL.Path, tequilapiUrlPrefix, "", 1)
+				req.URL.Path = strings.TrimRight(req.URL.Path, "/")
+			},
+			ModifyResponse: func(res *http.Response) error {
+				res.Header.Set("Access-Control-Allow-Origin", "*")
+				res.Header.Set("Access-Control-Allow-Methods", "POST, GET, OPTIONS, PUT, DELETE")
+
+				return nil
+			},
+		}
+		proxy.Transport = &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   24 * time.Hour,
+				KeepAlive: 24 * time.Hour,
+			}).DialContext,
+		}
+
+		// suppress SSE streaming errors
+		// there's nothing we can do about them
+		// and they don't cause any issues AFAIK
+		defer func() {
+			recover()
```

Comment:
this should handle the specific error instead.

Maybe something like this? Needs testing though.


```golang
defer func() {
			if err := recover(); err != nil {
				if err == http.ErrAbortHandler {
					// ignore
				} else {
					panic(err)
				}
			}
		}()
```